packages/wrangler/e2e/dev.test.ts

```diff
@@ -278,3 +280,91 @@ describe("dev registry", () => {
 		});
 	});
 });
+
+describe("hyperdrive dev tests", () => {
+	let worker: DevWorker;
+	let server: nodeNet.Server;
+
+	beforeEach(async () => {
+		worker = await makeWorker();
+		server = nodeNet.createServer().listen();
+		let port = 5432;
+		if (server.address() && typeof server.address() !== "string") {
+			port = (server.address() as nodeNet.AddressInfo).port;
+		}
+		await worker.seed((workerName) => ({
+			"wrangler.toml": dedent`
+					name = "${workerName}"
+					main = "src/index.ts"
+					compatibility_date = "2023-10-25"
+					
+					[[hyperdrive]]
+					binding = "HYPERDRIVE"
+					id = "hyperdrive_id"
+					localConnectionString = "postgresql://user:pass@127.0.0.1:${port}/some_db"
+			`,
+			"src/index.ts": dedent`
+					export default {
+						async fetch(request, env) {
+							if (request.url.includes("connect")) {
+								const conn = env.HYPERDRIVE.connect();
+								await conn.writable.getWriter().write(new TextEncoder().encode("test string"));
+							}
+							return new Response(env.HYPERDRIVE?.connectionString ?? "no")
+						}
+					}`,
+			"package.json": dedent`
+					{
+						"name": "${workerName}",
+						"version": "0.0.0",
+						"private": true
+					}
+					`,
+		}));
+	});
+
+	it("matches expected configuration parameters", async () => {
+		await worker.runDevSession("", async (port) => {
+			const { text } = await retry(
+				(s) => {
+					return s.status !== 200;
+				},
+				async () => {
+					const resp = await fetch(`http://127.0.0.1:${port}`);
+					return { text: await resp.text(), status: resp.status };
+				}
+			);
+			const url = new URL(text);
+			expect(url.pathname).toBe("/some_db");
+			expect(url.username).toBe("user");
+			expect(url.password).toBe("pass");
+			expect(url.host).not.toBe("localhost");
+		});
+	});
+
+	it("connects to a socket", async () => {
+		server.on("connection", (sock) => {
+			sock.on("data", (data) => {
+				expect(new TextDecoder().decode(data)).toBe("test string");
```

Comment:
Will this `expect` definitely run? What if the test exits too soon?

Reply:
This may need wrapping in a `new Promise()` that's explicitly await in the test

Reply:
Ah yeah, that makes sense. Just wrapped it in a promise